### PR TITLE
CV caching

### PR DESCRIPTION
**Pull request by supersergiy**
Implemented a `ChachedCloudVolume` class. Using `cachetools` to save up to 500 least recently used CV's.

### CODE REVIEW

File: corgie/mipless_cloudvolume.py [hunk 1]
```diff
@@ -1,12 +1,34 @@
...
+        result += '_'
+    return result
+
+class CachedCloudVolume(CloudVolume):
+    @cachetools.cached(
+        cachetools.LRUCache(maxsize=500),
+        key=jsonize_key
+    )
+    def __new__(self, *kargs, **kwargs):
+        print ("CREATING NEW CV")
```
Comment (nkemnitz) [inline corgie/mipless_cloudvolume.py]: corgie_logger.debug (or remove if just meant for testing)

Comment (supersergiy) [reply corgie/mipless_cloudvolume.py]: word. removed

File: corgie/mipless_cloudvolume.py [hunk 2]
```diff
@@ -1,12 +1,33 @@
...
 
 from corgie.log import logger as corgie_logger
 
+def jsonize_key(*kargs, **kwargs):
+    result = ''
+    for k in kargs[1:]:
+        result += json.dumps(k)
+        result += '_'
+
+    for k, v in kwargs.items():
```
Comment (nkemnitz) [inline corgie/mipless_cloudvolume.py]: I think you can simplify this a bit with `result += json.dumps(kwargs.items(), sort_keys=True)`. The `sort_keys=True` will also ensure that the order of keyword arguments (and keys of any nested dictionaries within kwargs) doesn't matter.

Comment (supersergiy) [reply corgie/mipless_cloudvolume.py]: Word. Made the changes

File: corgie/mipless_cloudvolume.py [hunk 3]
```diff
@@ -1,12 +1,31 @@
...
 
 from corgie.log import logger as corgie_logger
 
+def jsonize_key(*kargs, **kwargs):
+    result = ''
+    for k in kargs[1:]:
+        result += json.dumps(k)
+        result += '_'
+
+    json.dumps(kwargs.items(), sort_keys=True)
```
Comment (nkemnitz) [inline corgie/mipless_cloudvolume.py]:
```suggestion
    result += json.dumps(kwargs.items(), sort_keys=True)
```

Comment (supersergiy) [reply corgie/mipless_cloudvolume.py]: image

**Comment by supersergiy** [comment]
shouldn't write code before coffee

**Pull request merged**